src/SchemaNormalizer.php

```diff
@@ -230,6 +230,16 @@ private function normalizeObject(
 
         $valueSchema = new \stdClass();
 
+        /**
+         * @see https://json-schema.org/understanding-json-schema/reference/object.html#additional-properties
+         */
+        if (
+            \property_exists($schema, 'additionalProperties')
+            && \is_object($schema->additionalProperties)
+        ) {
+            $valueSchema = $schema->additionalProperties;
+        }
```

Comment:
@fredden 

I think this solves the problem better, what do you think?

Reply:
Yes, this looks like a much cleaner solution. Thanks for working through this. :+1: